### ISSUE TITLE
Editing certain notes locks interface for 10 sec or more, nothing in log?

### ISSUE THREAD

**Issue by LorenAmelang** [zadam/trilium#1595]
App version: | 0.45.8

Occasionally I find a (so far only web clipped) note that responds _very_ slowly to edit actions. Add or delete even one character and the Trilium interface is unresponsive for 10 to 15 seconds. Most of those I just give up on editing, and they function OK otherwise. 

But a few, especially the ones with the "Edit" box at top right, hide the entire note content while paralyzed, and stay locked for up to a minute, hogging maybe half of at least two CPUs. 

(Side question - why do some get the "Edit" box and others not?) 

Today's episode is in the attached log file, with my notes. I had clipped a page and opened it in Trilium:

--> I wanted to delete the full-note-width view of the /api/images/FdllMiZTGgIL/pdf.svg icon. No "Edit" box visible yet, so I clicked the image - several times in different places - no response. 

--> The "Edit" icon appeared, so I clicked that. All note content disappeared, and Trilium was paralyzed, using about half of two CPUs! Totally unresponsive. 

--> One of the things I clicked looking for any response was the close X for the problem note. When Trilium recovered, it closed the note. 

--> I re-opened it: 

--> Tried clicking Edit again...  

--> The note content disappeared, and Trilium was paralyzed - opened log...

--> Now the note is visible again, with no Edit box, and the pdf icon is icon sized instead of full width. (I was trying to delete the whole image, never saw the image size controls appear, but maybe they were active while invisible and I happened to click the "small" icon?)

I see no clues in the log about this...  Is there a way to turn on more detailed logging? 

[Trilium lockup.txt](https://github.com/zadam/trilium/files/5898854/Trilium.lockup.txt)

I can edit the note now, but every move takes the 10 - 15 seconds. Is there something I can do to get more clues when this happens?

**Comment by zadam** [comment]
Hi,

so the editing component (CKEditor) does not work well with huge notes. That's why you see "edit" button for long notes as a form of optimization. See [https://github.com/zadam/trilium/wiki/Text-notes](https://github.com/zadam/trilium/wiki/Text-notes) for some more info.

You mentioned this happens mostly with clipped notes. It would be helpful if you could provide a URL of such clipped webpage so I could analyze what's exactly happening.

**Comment by LorenAmelang** [comment]
The page in that log and report is: 
#clipType=page #pageUrl="https://academic.oup.com/edrv/article/25/5/693/2355205"  
https://academic.oup.com/edrv/article/25/5/693/2355205

I guess it is long - 376 references...  But it doesn't seem to me the pauses in responsiveness correlate with simple file length. Is page complexity a factor? Maybe "Note Info" could include the size or complexity or whatever parameter triggers the "Edit" box? 

Could there be a "wait cursor" to signify the task is just slow and the program has not totally crashed?

Is there some more appropriate way I should be clipping "long" academic papers? The whole point is to be able to full text search them for some new concept, long after I've forgotten what all they covered. So it seems I need the full text...

**Comment by zadam** [comment]
Hi, thanks for the link.

> I guess it is long - 376 references... But it doesn't seem to me the pauses in responsiveness correlate with simple file length. Is page complexity a factor?

Yes, it is long. RAW HTML has 1.6 MB, after simplification and cleanup 310 KB. That's a lot of text.

HTML complexity plays a role, but what exactly is difficult to say. The decision to display "Edit" is based on HTML length only - and cutoff is currently at 10 KB.

> Could there be a "wait cursor" to signify the task is just slow and the program has not totally crashed?

Unfortunately no. CKEditor run synchronously which is faster, but also means it essentially freezes the browser while it's working (JS is single thread).

> Is there some more appropriate way I should be clipping "long" academic papers? The whole point is to be able to full text search them for some new concept, long after I've forgotten what all they covered. So it seems I need the full text...

If you need the text to be editable then no. CKEditor is slower than I would hope and does not handle long texts well. There might be improvements in the future in this regard but there probably won't be any miracles.

If you need such long editable notes then Trilium is probably not a good choice.

**Comment by LorenAmelang** [comment]
Thanks, Zadam! I really appreciate your attention to such details. 

Editable is nice, but I guess I could link to a "comments" note with my additions. So clip like now and just don't ever use Edit...  

As for alternatives, I can't justify the price of Roam, so haven't even tried it. Trilium far outperforms everything else, for my purposes! I tried most of them...  Once I've touched up a clipped page to my liking, the search is amazing.
